### PR TITLE
Add data ingestion

### PR DESCRIPTION
**Pull request by s-barrah**
Type: patch

Fixes https://github.com/comicrelief/comicrelief/issues/792

## Changes proposed in this PR

- Send ESU data to the data ingestion pipeline

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -107,6 +107,7 @@
         }
     },
     "require": {
+        "ext-json": "*",
```
Comment (AndyEPhipps) [inline composer.json]: Maybe better to lock this down to a minor version at least?

File: modules/cr_email_signup/src/DataIngestion/ESUDataIngestion.php [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+namespace Drupal\cr_email_signup\DataIngestion;
+
+
+/**
+ * Email Sign up Data Ingestion class.
+ */
+class ESUDataIngestion
+{
+
+    protected $data_ingestion_endpoint = 'https://ingest-staging.data.comicrelief.com/identify';
```
Comment (AndyEPhipps) [inline modules/cr_email_signup/src/DataIngestion/ESUDataIngestion.php]: Might be better setting this as a Drupal config setting in `cr_email_signup_settings.yml`, that way we can set-up different endpoints for dev and prod within our CR/SR projects

Comment (s-barrah) [reply modules/cr_email_signup/src/DataIngestion/ESUDataIngestion.php]: Sounds good

File: composer.json (hunk 1 above)
Comment (adamclark-dev) [reply composer.json]: Agree here, let's lock this down

Comment (s-barrah) [reply composer.json]: Doesn't work

**Comment by adamclark-dev** [comment]
@s-barrah tests not passing

**Pull request closed without merging**